messaging.wit.md

```diff
@@ -0,0 +1,105 @@
+# `wasi-messaging` API
+
+## Interfaces
+
+```wit
+/// An interface grouping all necessary types for messaging.
+interface "wasi:messaging/types" {
+    /// An event type that follows the CloudEvents specification (https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md). We
+    /// assume the type of the data is a byte sequence. It is up to the data schema to determine what type of the data payload the event 
+    /// contains.
+    record event {
+    	specversion: string,
+    	ty: string,
+    	source: string,
+    	id: string,
+    	data: option<list<u8>>,
+    	datacontenttype: option<string>,
+    	dataschema: option<string>,
+    	subject: option<string>,
+    	time: option<string>,
+    }
```

Comment:
I really like that this uses the CloudEvents standard!
How could we add support for [extension attributes](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#extension-context-attributes) ?
Should there be another field `extensions: option<map<string,string>>` ?

Reply:
Thank you for bringing this up~ I'll make a commit including that!